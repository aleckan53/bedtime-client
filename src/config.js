export default {
  API_ENDPOINT: process.env.API_ENDPOINT || 'https://bedtime-server.herokuapp.com/api',
  TOKEN_NAME: process.env.TOKEN_NAME || 'bedtime_app'
}