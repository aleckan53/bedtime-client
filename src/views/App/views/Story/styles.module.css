.Story img {
  width: 100%;
}

.Story {
  text-align: justify;
  line-height: 1.7;
}

.Story h1 {
  text-align: center;
  margin: 3rem 0;
}

.Story p {
  font-size: 1.2rem;
  margin-bottom: 2.5rem;
}

.unauth h1 {
  text-align: center;
}

.top {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.top p {
  font-size: .8rem;
  padding: 0;
  margin: 0;
}

.top h3 {
  font-size: .8rem;
  padding: 0;
  margin: 0;
  text-align: right;
}

.unauth a {
  text-decoration: none;
  border: 1px solid black;
  width: 100%;
  display: block;
  margin: 1rem 0;
  box-sizing: border-box;
  height: 2rem;
  text-align: center;
  padding: .35rem 0;
  color: black;
  border-radius: 20px
}

.unauth img {
  height: 400px;
  width: 100%;
  border-radius: 20px;
  object-fit: cover;
}

.controls {
  width: 100%;
  height: 3rem;
  border-radius: 10px;
  border: 1px solid black;
  margin: 2rem 0;
  box-sizing: border-box;
  font-size: 2rem;
  display: flex;
  justify-content: space-around;
  align-items: center;
  opacity: .1;
  transition: all 1s ease;
  position: sticky;
  top: 20px;
}

.controls:hover {
  opacity: .9;
}

.controls svg {
  
}