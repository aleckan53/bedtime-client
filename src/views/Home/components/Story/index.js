import React, { useState, useEffect } from 'react'
import Api from './api'

const Story = props => {

  const [story, setStory] = useState({})

  useEffect(() => {
    Api.getStoryById(props.match.params.id)
      .then(setStory)

    return () => {
      setStory({})
    }
  }, [props.match.params.id])
  
  return (
    <div>
      <h1>{story.name}</h1>
      <h3>Written by: {story.author}</h3>
      <div>
        {story.content}
      </div>
    </div>
  )
}

export default Story