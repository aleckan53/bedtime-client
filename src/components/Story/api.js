import config from 'config'

export default {
  getStoryById(id) {
    return fetch(`${config.API_ENDPOINT}/stories/${id}`, {
      headers: {
        'Authorization': `Bearer ${window.sessionStorage.getItem(config.TOKEN_NAME)}`
      }
    })
      .then(res => !res.ok ? res.json().then(e => Promise.reject(e)) : res.json())
      .catch(console.error)
  }
}