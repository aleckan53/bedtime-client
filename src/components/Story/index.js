import React, { useState, useEffect } from 'react'
import Api from './api'
import styles from './styles.module.css'
import config from 'config'
import { Link } from 'react-router-dom'
import { 
  IoIosArrowUp as Up
} from 'react-icons/io'

const Story = props => {

  const hasToken = !!window.sessionStorage.getItem(config.TOKEN_NAME)
  const [story, setStory] = useState({content: ''})
  useEffect(() => {
    if(hasToken) {
      Api.getStoryById(props.match.params.id)
        .then(setStory)
    }

    return () => {
      setStory({})
    }
  }, [props.match.params.id, hasToken])

  const { link, name } = props.stories.find(s => s.id === Number(props.match.params.id)) || {
    link: '',
    name: ''
  }

  const calcReadTime = content => {
    return Math.ceil(content.length / 850)
  }

  return !hasToken ? (
    <section className={styles.unauth}>
      <h1>Please login or signup to read the story</h1>
      <div className={styles.imgWrapper}>
        <img src={link} alt={name}/>
        <h2>{name}</h2>
      </div>
      <div>
        <Link to='/app/auth'>Login / Signup</Link>
        <Link to='/app/home'>Back to home page</Link>
      </div>
    </section>
    ) : (
    <section className={styles.Story}>
      <button className={styles.up} onClick={() => {
          window.scrollTo({
            top: 100,
            left: 100,
            behavior: 'smooth'
          })
        }}>
        <Up/>
      </button>
      <div className={styles.top}>
        <p>{calcReadTime(story.content)} min read time</p>
        <h3>Author: {story.author}</h3>
      </div>
      <h1>{story.name}</h1>
      <div dangerouslySetInnerHTML={{__html: story.content}}>
      </div>
    </section>
    )
}

export default Story