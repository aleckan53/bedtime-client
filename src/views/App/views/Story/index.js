import React, { useState, useEffect } from 'react'
import Api from './api'
import styles from './styles.module.css'
import config from 'config'
import { Link } from 'react-router-dom'
import { 
  IoIosBookmark as Bmrk,
  IoIosVolumeHigh as Voice,
  IoMdSunny as Light,
  IoIosMoon as Dark
} from 'react-icons/io'

const Story = props => {

  const [story, setStory] = useState({content: ''})

  useEffect(() => {
    Api.getStoryById(props.match.params.id)
      .then(setStory)
    return () => {
      setStory({})
    }
  }, [props.match.params.id])

  const { link, name } = props.stories.find(s => s.id === Number(props.match.params.id)) || {
    link: '',
    name: ''
  }
  const hasToken = !!window.sessionStorage.getItem(config.TOKEN_NAME)

  const calcReadTime = content => {
    return Math.ceil(content.length / 850)
  }

  return !hasToken ? (
    <section className={styles.unauth}>
      <h1>Please login or signup to read the story</h1>
      <img src={link} alt={name}/>
      <div>
        <Link to='/app/auth'>Login / Signup</Link>
        <Link to='/app/home'>Back to home page</Link>
      </div>
    </section>
    ) : (
    <section className={styles.Story}>
      <div className={styles.controls}>
        <Light/>
        <Dark/>
        <Voice/>
        <Bmrk/>
      </div>
      <div className={styles.top}>
        <p>{calcReadTime(story.content)} min read time</p>
        <h3>Author: {story.author}</h3>
      </div>
      <h1>{story.name}</h1>
      <div dangerouslySetInnerHTML={{__html: story.content}}>
      </div>
    </section>
    )
}

export default Story