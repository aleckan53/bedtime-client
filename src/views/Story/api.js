import axios from 'axios'
import config from 'config'

export default {
  getStoryById(id) {
    return axios.get(`${config.API_ENDPOINT}/stories/${id}`, {
      headers: {
        'Authorization': `Bearer ${window.sessionStorage.getItem(config.TOKEN_NAME)}`
      }
    })
      .then(res => res.data)
      .catch(console.error)
  }
}