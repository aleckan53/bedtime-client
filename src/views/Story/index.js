import React, { useState, useEffect } from 'react'
import Api from './api'
import styles from './styles.module.css'
import config from 'config'
import { Link } from 'react-router-dom'

const Story = props => {

  const [story, setStory] = useState({})

  useEffect(() => {
    Api.getStoryById(props.match.params.id)
      .then(setStory)
    return () => {
      setStory({})
    }
  }, [props.match.params.id])

  const { link, name } = props.stories.find(s => s.id === Number(props.match.params.id)) || {
    link: '',
    name: ''
  }
  const hasToken = !!window.sessionStorage.getItem(config.TOKEN_NAME)
  return !hasToken ? (
    <section className={styles.unauth}>
      <h1>Please login or signup to read the story</h1>
      <img src={link} alt={name}/>
      <div>
        <Link to='/auth'>Login / Signup</Link>
        <Link to='/home'>Back to home page</Link>
      </div>
    </section>
    ) : (
    <section className={styles.Story}>
      <p>Estimated read time: {`1 min`}</p>
      <h1>{story.name}</h1>
      <h3>Written by: {story.author}</h3>
      <div dangerouslySetInnerHTML={{__html: story.content}}>
      </div>
    </section>
    )
}

export default Story