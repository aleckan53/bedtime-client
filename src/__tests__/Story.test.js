import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Story from '../components/Story'
import { BrowserRouter } from 'react-router-dom'
import config from 'config'

describe(`Story Component`, () => {
  const stories = [
    {id: 1, name: 'test', rating: 5},
    {id: 12, name: 'test2', rating: 3},
  ]

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Story match={{params: {id: 1}, isExact: true, path: "", url: ""}}  stories={stories}/>
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })
  describe('Unauthorized', () => {
    it('matches the snapshot', () => {
      const wrapper = shallow(<Story stories={stories} match={{params: {id: 1}, isExact: true, path: "", url: ""}}/>)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
  describe('Authorized', () => {
    it('matches the snapshot', () => {
      window.sessionStorage.setItem(config.TOKEN_NAME, 'test')

      const wrapper = shallow(<Story stories={stories} match={{params: {id: 1}, isExact: true, path: "", url: ""}}/>)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})