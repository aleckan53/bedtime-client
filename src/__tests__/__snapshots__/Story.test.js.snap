// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Component Authorized matches the snapshot 1`] = `
<section
  className="Story"
>
  <button
    className="up"
    onClick={[Function]}
  >
    <IoIosArrowUp />
  </button>
  <div
    className="top"
  >
    <p>
      0
       min read time
    </p>
    <h3>
      Author: 
    </h3>
  </div>
  <h1 />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</section>
`;

exports[`Story Component Unauthorized matches the snapshot 1`] = `
<section
  className="unauth"
>
  <h1>
    Please login or signup to read the story
  </h1>
  <div
    className="imgWrapper"
  >
    <img
      alt="test"
    />
    <h2>
      test
    </h2>
  </div>
  <div>
    <Link
      to="/app/auth"
    >
      Login / Signup
    </Link>
    <Link
      to="/app/home"
    >
      Back to home page
    </Link>
  </div>
</section>
`;
